Source/mpqapi.cpp

```diff
@@ -666,11 +628,13 @@ BOOLEAN mpqapi_write_header() // WriteMPQHeader
 	fhdr.blockoffset = 104;
 	fhdr.hashcount = 2048;
 	fhdr.blockcount = 2048;
-	if (SetFilePointer(sghArchive, 0, NULL, FILE_BEGIN) != -1 && WriteFile(sghArchive, &fhdr, 0x68u, &NumberOfBytesWritten, 0))
-		result = NumberOfBytesWritten == 104;
-	else
-		result = 0;
-	return result;
+
+	if (SetFilePointer(sghArchive, 0, NULL, FILE_BEGIN) == -1)
+		return 0;
+	if (!WriteFile(sghArchive, &fhdr, 0x68, &NumberOfBytesWritten, 0))

```

Comment:
Likewise, if `sizeof(_FILEHEADER)` is 0x68, then:

```suggestion
	if (!WriteFile(sghArchive, &fhdr, sizeof(_FILEHEADER), &NumberOfBytesWritten, 0))
```